src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -573,7 +573,7 @@ public function mergeCasts($casts)
      * @param  mixed  $value
      * @return mixed
      */
-    protected function castAttribute($key, $value)
+    public function castAttribute($key, $value)
```

Comment:
Is there a need for this? Why does this method need to be `public`? This seems to me like an unnecessary breaking change.

Reply:
I will restore it as protected. Perhaps it can be extended in the future with something public.

Using cast looks like a good way to control types in strongly-typed databases.

Currently, there is no way to access cast method if for example user would like to do something like `where('user_id', $user->castAttribute('id', $user->id))`.